### DISCUSSION TITLE
Define $_Server[DOCUMENT_ROOT]

### DISCUSSION THREAD

**Discussion by jcvignoli** [vimeo/psalm#8587] [Q&A] [answered]
Hi there,

I'm trying to declare the global constant  $_SERVER['DOCUMENT_ROOT'] but I can't. Psalm seems to not be able to find included files that I call using that variable.
I created a bootstrap file called in psalm.xml's autolader option:

`$_SERVER['DOCUMENT_ROOT'] ="./src";`

But psalm continues to complain that files (well, declared functions in files) are not found.

Hiw am I supposed to manage psalm settings so global constant such as $_SERVER['DOCUMENT_ROOT'] is found?

Thanks.

**Comment by orklah** [comment, marked as answer]
you could try to document $_SERVER in https://psalm.dev/docs/running_psalm/configuration/#globals. Not sure we ever had this issue reported before though. It may not be supported at all.

**Reply by jcvignoli** [reply]
Thanks, it helped me to solve a few issues.
I understand it's an unnusual question. But I struggle finding a way Psalm succeed with the includes in my files. I haven't been able so far to find an acceptable way for simple includes.
Thanks!
